sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -169,73 +169,106 @@ inline pi_result ur2piInfoValue(ur_device_info_t ParamName,
   ConvertHelper Value(ParamValueSizePI, ParamValue, ParamValueSizeUR);
 
   if (ParamName == UR_DEVICE_INFO_TYPE) {
-    static std::unordered_map<ur_device_type_t, pi_device_type> Map = {
-        {UR_DEVICE_TYPE_CPU, PI_DEVICE_TYPE_CPU},
-        {UR_DEVICE_TYPE_GPU, PI_DEVICE_TYPE_GPU},
-        {UR_DEVICE_TYPE_FPGA, PI_DEVICE_TYPE_ACC},
+    auto ConvertFunc = [](ur_device_type_t UrValue) {
+      switch (UrValue) {
+      case UR_DEVICE_TYPE_CPU:
+        return PI_DEVICE_TYPE_CPU;
+      case UR_DEVICE_TYPE_GPU:
+        return PI_DEVICE_TYPE_GPU;
+      case UR_DEVICE_TYPE_FPGA:
+        return PI_DEVICE_TYPE_ACC;
+      default:
+        die("ConvertHelper: unhandled value");
+      }
     };
-    return Value.convert(Map);
+    return Value.convert<ur_device_type_t, pi_device_type>(ConvertFunc);
   } else if (ParamName == UR_DEVICE_INFO_QUEUE_PROPERTIES) {
-    static std::unordered_map<ur_queue_flag_t, pi_queue_properties> Map = {
-        {UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE,
-         PI_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE},
-        {UR_QUEUE_FLAG_PROFILING_ENABLE, PI_QUEUE_FLAG_PROFILING_ENABLE},
-        {UR_QUEUE_FLAG_ON_DEVICE, PI_QUEUE_FLAG_ON_DEVICE},
-        {UR_QUEUE_FLAG_ON_DEVICE_DEFAULT, PI_QUEUE_FLAG_ON_DEVICE_DEFAULT},
+    auto ConvertFunc = [](ur_queue_flag_t UrValue) {
+      switch (UrValue) {
+      case UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE:
+        return PI_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+      case UR_QUEUE_FLAG_PROFILING_ENABLE:
+        return PI_QUEUE_FLAG_PROFILING_ENABLE;
+      case UR_QUEUE_FLAG_ON_DEVICE:
+        return PI_QUEUE_FLAG_ON_DEVICE;
+      case UR_QUEUE_FLAG_ON_DEVICE_DEFAULT:
+        return PI_QUEUE_FLAG_ON_DEVICE_DEFAULT;
+      default:
+        die("ConvertHelper: unhandled value");
+      }
     };
-    return Value.convertBitSet(Map);
+    return Value.convertBitSet<ur_queue_flag_t, pi_queue_properties>(
+        ConvertFunc);
   } else if (ParamName == UR_DEVICE_INFO_EXECUTION_CAPABILITIES) {
-    static std::unordered_map<ur_device_exec_capability_flag_t,
-                              pi_queue_properties>
-        Map = {
-            {UR_DEVICE_EXEC_CAPABILITY_FLAG_KERNEL,
-             PI_DEVICE_EXEC_CAPABILITIES_KERNEL},
-            {UR_DEVICE_EXEC_CAPABILITY_FLAG_NATIVE_KERNEL,
-             PI_DEVICE_EXEC_CAPABILITIES_NATIVE_KERNEL},
-        };
-    return Value.convertBitSet(Map);
+    auto ConvertFunc = [](ur_device_exec_capability_flag_t UrValue) {
+      switch (UrValue) {
+      case UR_DEVICE_EXEC_CAPABILITY_FLAG_KERNEL:
+        return PI_DEVICE_EXEC_CAPABILITIES_KERNEL;
+      case UR_DEVICE_EXEC_CAPABILITY_FLAG_NATIVE_KERNEL:
+        return PI_DEVICE_EXEC_CAPABILITIES_NATIVE_KERNEL;
+      default:
+        die("ConvertHelper: unhandled value");

```

Comment:
NIT: Since we have multiple differrent die now we can give a more specific error message, e.g.
 ```suggestion
         die("UR_DEVICE_INFO_EXECUTION_CAPABILITIES: unhandled value");
```

Reply:
Good idea, done